### PR TITLE
add image modes

### PR DESCRIPTION
**Pull request by foolab**
These series of commits add image-mode and supported-image-modes properties.
This enables 4 modes:
Normal image capture which is what we have been using so far
ZSL
HDR
ZSL and HDR at the same time

### CODE REVIEW

File: gst/droidcamsrc/gstdroidcamsrc.c [hunk 1]
```diff
@@ -2207,3 +2226,53 @@ gst_droidcamsrc_find_picture_resolution (GstDroidCamSrc * src,
...
+gboolean
+gst_droidcamsrc_is_hdr_enabled (GstDroidCamSrc * src)
+{
+  return src->image_mode & GST_DROIDCAMSRC_IMAGE_MODE_HDR
+      || src->image_mode & GST_DROIDCAMSRC_IMAGE_MODE_ZSL_AND_HDR;
+}
+
+
+static gboolean
+gst_droidcamsrc_is_zsl_and_hdr_supported (GstDroidCamSrc * src)
```
Comment (saukko) [inline gst/droidcamsrc/gstdroidcamsrc.c:2241]: I feel that these should really be something like:
gst_droidcamsr_supported_modes() or something that would return array what is supported.

And similar thing for above gst_droidcamsr_enabled_modes() that would return what is enabled.

As if there will be more modes available in the future these will not really scale if you need to do separate functions for:
a_and_b
b_and_c
a_and_c
a_and_b_and_c
...

Comment (foolab) [reply gst/droidcamsrc/gstdroidcamsrc.c:2241]: Those are internal implementation details. Not part of the API offering at all and can be refactored anytime without affecting the API at all

Comment (saukko) [reply gst/droidcamsrc/gstdroidcamsrc.c:2241]: Not part of api offering, what does that mean? Nothing outside gst-droid uses it?

Comment (foolab) [reply gst/droidcamsrc/gstdroidcamsrc.c:2241]: The functions are implementation detail.
The 2 properties image-mode and supported-image-modes are the new additions to be used for interacting with upper layers.

**Comment by abranson** [comment]
LGTM!

**Pull request merged**